ovve_ui/utils/comms_link.py

```diff
@@ -229,20 +238,35 @@ def process_SerialData(self) -> None:
 
             self.settings_lock.release()
 
-    def init_serial(self):
-    
-        #global ser          #Must be declared in Each Function
+    def init_serial(self) -> False:
+
         self.ser = serial.Serial()
         self.ser.baudrate = self.BAUD
         self.ser.port = self.PORT 
         #a little less than polling spped from arduino
         self.ser.timeout = self.SER_TIMEOUT
-        self.ser.open()          #Opens SerialPort
+        
+        try:
+            
+            if self.ser == None:
+                self.ser.open()
+                print ("Successfully connected to port %r." % self.ser.port)
+                return True
+            else:
+                if self.ser.isOpen():
```

Comment:
Do you want to try to open it again after you close the existing connection here?

Reply:
I have to fix this later because , when there is no cable i- I cannot trap the exception. Still researching.